lib/galaxy/config/__init__.py

```diff
@@ -162,6 +172,78 @@ def _set_config_directories():
         _set_global_conf()
         _set_config_directories()
 
+    def _load_schema(self):
+        # Override in subclasses
+        raise Exception('Not implemented')
+
+    def _deprecated_dirs(self):
+        # Override in subclasses (optional)
+        # {KEY: config option, VALUE: deprecated directory name}
+        # If VALUE == first directory in a user path that resolves to KEY, it will be stripped from the path
+        return None
```

Comment:
I'd keep `deprecated_dirs` as a dict, which here in `BaseAppConfiguration` would just be `{}`.

Reply:
Done.